nupic/data/record_stream.py

```diff
@@ -25,178 +25,271 @@
 import datetime
 
 
+from nupic.data.fieldmeta import FieldMetaSpecial
 
-class RecordStreamIface(object):
-  """This is the interface for the record input/output storage classes."""
 
-  __metaclass__ = ABCMeta
 
+def _getFieldIndexBySpecial(fields, special):
+  """ Return index of the field matching the field meta special value.
+  :param fields: sequence of nupic.data.fieldmeta.FieldMetaInfo objects
+    representing the fields of a stream
+  :param special: one of the special field attribute values from
+    nupic.data.fieldmeta.FieldMetaSpecial
+  :returns: first zero-based index of the field tagged with the target field
+    meta special attribute; None if no such field
+  """
+  for i, field in enumerate(fields):
```

Comment:
@oxtopus: What happened to duck-typing, anyway? :) This is a private method. There are multiple places where the fields are passed and used, so this is probably not the ideal place for it. Public methods are usually where we might have checks like that.